DSL/conditions.md

```diff
@@ -0,0 +1,237 @@
+# The Condition DSL
+
+This document provides an overview of a declarative Domain Specific Language (DSL) designed to check various conditions for upgrade and downgrade rules in this repository.
+
+## Why Not Use JSON Schema Validation?
+
+There are several reasons for opting against JSON Schema as the schema language for these conditions:
+
+- **Simplicity**: Implementers do not need a full-blown JSON Schema implementation to support upgrade/downgrade rules in their language of choice. This reduces the complexity and learning curve for developers.
+- **Performance**: JSON Schema validation can be slow, especially on Alterschema. By focusing on a subset of operations, we achieve much better performance.
+- **Specific Needs**: The modifications needed are typically only 1-2 levels deep, and only 20-30% of JSON Schema constraints are necessary to satisfy the conditions for upgrades/downgrades.
+
+## DSL Structure
+
+The DSL is structured as an array of conditions, with each condition containing three mandatory fields: `path`, `operation`, and `value`.
+
+### Condition Fields
+
+- `path`: Specifies the path to the target in the schema where the operation will be performed. It takes an array. Each element of the array can be:
```

Comment:
I wonder if this might make more sense as a table. i.e. a table explaining the field names, types, and short description, and having the bigger pattern example after the table